### PR TITLE
Adding type constraints to variables.

### PR DESCRIPTION
**Pull request by lpena-recurly**
For best practices and for terragrunt users [0]. Type constrains should be added [1]. Without specifying these constraints, applying a terragrunt plan/apply results in type errors [2], depending on the type constraints.

In terragrunt, "values are being passed in with environment variables and json, the type information is lost when crossing the boundary between Terragrunt and Terraform. You must specify the proper type constraint on the variable in Terraform in order for Terraform to process the inputs to the right type" [3]. Similar PR [4]

tl;dr lists and maps currently do not work with terragrunt because type constraints are not explicitly set.

[0] https://github.com/gruntwork-io/terragrunt
[1] https://www.terraform.io/docs/configuration/variables.html#type-constraints
[2] Invalid value for "inputMap" parameter: lookup() requires a map as the first
argument.
[3] https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/#inputs
[4] https://github.com/terraform-google-modules/terraform-google-cloud-storage/pull/43

### CODE REVIEW

File: modules/fabric-net-firewall/variables.tf [hunk 1]
```diff
@@ -16,24 +16,29 @@
...
 
 variable "internal_ranges" {
   description = "IP CIDR ranges for intra-VPC rules."
+  type        = list(string)
   default     = []
 }
 
 variable "internal_allow" {
   description = "Allow rules for internal ranges."
+  type        = list(string)
```
Comment (morgante) [inline modules/fabric-net-firewall/variables.tf]: This is a list of objects actually.

Comment (lpena-recurly) [reply modules/fabric-net-firewall/variables.tf]: Great! got the fixes in and generated docs.

Comment (lpena-recurly) [reply modules/fabric-net-firewall/variables.tf]: since one parameter is optional, there is no way to work around that with only 1 defined parameter. Removing based off of hashicorp/terraform#19898 . This will not affect the functionality of the module.

Comment (lpena-recurly) [reply modules/fabric-net-firewall/variables.tf]: since one parameter is optional, there is no way to work around that with only 1 defined parameter. Removing based off of hashicorp/terraform#19898 . This will not affect the functionality of the module.

**Comment by morgante** [comment]
@lpena-recurly Looks like tests are still failing. Could you get linters passing?

File: modules/fabric-net-firewall/variables.tf [hunk 2]
```diff
@@ -16,53 +16,59 @@
...
 
 variable "internal_allow" {
   description = "Allow rules for internal ranges."
-  default = [
-    {
-      protocol = "icmp"
-    },
-  ]
+  type        = list(object({ protocol = string, ports = list(string) }))
+  default     = []
```
Comment (morgante) [inline modules/fabric-net-firewall/variables.tf]: Please don't change the defaults.

Comment (lpena-recurly) [reply modules/fabric-net-firewall/variables.tf]: got it. will resolve this soon. The linters would not pass because I changed the default. But the current default wont allow terragrunt to work. So I am looking for another solution.

**Comment by morgante** [comment]
@lpena-recurly Are you still planning to work on this?

File: modules/fabric-net-firewall/variables.tf (hunk 2 above)
Comment (forest) [reply modules/fabric-net-firewall/variables.tf:42]: I had the same issue, and used this as the default. It does change the default but makes terragrunt happy.

```
default = [
    {
      protocol = "icmp"
      ports    = []
    },
  ]
```

I'm not sure what the correct solution is.

Comment (morgante) [reply modules/fabric-net-firewall/variables.tf:42]: Does Terragrunt work with `type = list(any)`?

**Comment by lpena-recurly** [comment]
@morgante Apologies for the delay. yes I will be fixing this soon.

**Comment by morgante** [comment]
I'm closing this. @lpena-recurly please reopen if/when you can fix conflicts.
